draft-ietf-httpbis-http2-secondary-certs.md

```diff
@@ -425,18 +427,32 @@ Client                                      Server
    -- (stream 0) CERTIFICATE ------------------>
    -- (stream N) USE_CERTIFICATE -------------->
    <----------------------- (stream N) 200 OK --
-
 ~~~
 {: #ex-http2-client-requested title="Reactive Certificate Authentication"}
 
-A server SHOULD provide certificates for an origin before pushing resources from
-it or supplying content referencing the origin. If a client receives a
-`PUSH_PROMISE` referencing an origin for which it has not yet received the
-server's certificate, the client MUST verify the server's possession of an
-appropriate certificate by sending a `CERTIFICATE_NEEDED` frame on the pushed
-stream to inform the server that progress is blocked until the request is
-satisfied. The client MUST NOT use the pushed resource until an appropriate
-certificate has been received and validated.
+On subsequent requests, the client can proactively indicate that an existing
+certificate MAY be used by including a `USE_CERTIFICATE` frame as part of the
+request.  If the indicated certificate does not correspond to the request the
```

Comment:
You need to explain what ordering you expect to have here.  After the header block, but before the request body.  Which implies an empty DATA frame rather than a HEADERS frame with FIN attached.